packages/cubejs-server-core/src/core/CompilerApi.js

```diff
@@ -229,22 +423,102 @@ export class CompilerApi {
     );
   }
 
-  async metaConfig(options = {}) {
+  /**
+   * if RBAC is enabled, this method is used to filter out the cubes that the
+   * user doesn't have access from meta responses.
+   * It evaluates all applicable memeberLevel accessPolicies givean a context
+   * and retains members that are allowed by any policy (most permissive set).
+  */
+  async filterVisibilityByAccessPolicy(cubeEvaluator, context, cubes) {
+    const isMemberVisibleInContext = {};
+
+    if (!cubeEvaluator.isRbacEnabled()) {
+      return cubes;
+    }
+
+    for (const cube of cubes) {
+      const evaluatedCube = cubeEvaluator.cubeFromPath(cube.config.name);
+      if (cubeEvaluator.isRbacEnabledForCube(evaluatedCube)) {
+        const applicablePolicies = await this.getApplicablePolicies(evaluatedCube, context, cubeEvaluator);
+
+        const computeMemberVisibility = async (item) => {
```

Comment:
It seems there is no need for async here
```suggestion
        const computeMemberVisibility = (item) => {
```

Reply:
agree
it was async until I moved `getApplicablePolicies` out :)